### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by IAlibay**
- Add in the shim for macos.

### CODE REVIEW

File: docs/environment.yaml [hunk 1]
```diff
@@ -9,4 +9,4 @@ dependencies:
 - sphinx
 - openmm
 - pip:
-  - git+https://github.com/OpenFreeEnergy/ofe-sphinx-theme@main
+  - git+https://github.com/OpenFreeEnergy/ofe-sphinx-theme@a45f3edd5bc3e973c1a01b577c71efa1b62a65d6
```
Comment (IAlibay) [inline docs/environment.yaml:12]: This is a known issue upstream

File: .github/workflows/mypy.yaml [hunk 2]
```diff
@@ -29,7 +29,8 @@ jobs:
           cache-environment: true
           cache-downloads: true
           create-args: >-
-            python=3.9
+            python=3.10
+            rdkit=2023.09.5
```
Comment (IAlibay) [inline .github/workflows/mypy.yaml:33]: I've never been able to work that one out.

Comment (dotsdl) [reply .github/workflows/mypy.yaml:33]: @IAlibay can you elaborate? Work what out here?

File: docs/environment.yaml (hunk 1 above)
Comment (dotsdl) [reply docs/environment.yaml:12]: Can you link to the issue? I'd like to learn what's going on here so we can follow up later.

**Review by dotsdl — APPROVED** [review]
@IAlibay some questions, but nothing blocking.

Do we know why some of the CI runs appear to be waiting endlessly?

File: .github/workflows/mypy.yaml (hunk 2 above)
Comment (IAlibay) [reply .github/workflows/mypy.yaml:33]: For some reason or another mypy has been complaining about a certain error with rdkit 2024+. Something about self being defined twice for a certain object.

Our approach has just been to pin to an older version of rdkit, but honestly we should try to fix it properly.

File: docs/environment.yaml (hunk 1 above)
Comment (IAlibay) [reply docs/environment.yaml:12]: Here we go: https://github.com/OpenFreeEnergy/ofe-sphinx-theme/issues/5

It's a fixaeble packaging issue but I've not had time to look into it.

**Comment by IAlibay** [comment]
> Do we know why some of the CI runs appear to be waiting endlessly?

It's because we've changed the matrix and some of the checks are enforced as "necessary" in the gufe repo settings, so github is waiting for runners that don't exist.

**Pull request merged**